### ISSUE TITLE
Selector equivalent

### ISSUE THREAD

**Issue by masterlonghorn** [rrousselGit/river_pod#511]
Couple of questions:

1) What is the equivalent of Provider's Selector in Riverpod?
2) Does StateNotifier support Maps, Lists etc.? If not, why is it more preferable over ChangeNotifier which lets you keep data organized in one place?

**Comment by rrousselGit** [comment]
At the moment `Selector` required flutter_hooks:

```dart
final isEven = useProvider(counterProvider.select((count) => count.isEven));
```

This feature is going to be available without hooks as part of the 1.0.0

**Comment by rrousselGit** [comment]
> 2\. Does StateNotifier support Maps, Lists etc.? If not, why is it more preferable over ChangeNotifier which lets you keep data organized in one place?

Yes it does.

StateNotifier is about using immutable models instead of mutable models.
This makes code more maintainable in the long run. 

In terms of organization, ChangeNotifier is not "more organized" than StateNotifier.
